db/version_set.cc

```diff
@@ -2060,6 +2074,17 @@ void Version::MultiGet(const ReadOptions& read_options, MultiGetRange* range,
           nullptr /* result_operand */, true);
       if (LIKELY(iter->value != nullptr)) {
         iter->value->PinSelf();
+        range->AddValueSize(iter->value->size());
+        if (range->GetValueSize() > read_options.value_size_soft_limit) {
```

Comment:
Can track an overall status and combine this block and the one in lines 2010-2016 into one at the end of the function?